sdk/helper/aliasmetadata/alias_metadata.go

```diff
@@ -0,0 +1,195 @@
+// regarding how to use the "alias_metadata" field.
+func FieldSchema(fields *Fields) *framework.FieldSchema {
+	return &framework.FieldSchema{
+		Type:        framework.TypeCommaStringSlice,
+		Description: description(fields),
+		DisplayAttrs: &framework.DisplayAttributes{
+			Name:  fields.FieldName,
+			Value: "default,field1,field2",
+		},
+		Default: []string{"default"},
+	}
+}
+
+// NewHandler instantiates a Handler to be embedded in your config.
+func NewHandler(fields *Fields) Handler {
+	return &handler{
+		fields: fields,
+	}
+}
+
+// Handler is an interface for the helper methods you get on your
+// config when you embed the Handler.
+type Handler interface {
+	GetAliasMetadata() []string
+	ParseAliasMetadata(data *framework.FieldData) error
+	PopulateDesiredAliasMetadata(auth *logical.Auth, fieldValues map[string]string) error
+}
+
+type handler struct {
+	// AliasMetadata is an explicit list of all the user's configured
+	// fields that are being added to alias metadata. It will never
+	// include the "default" parameter, and instead includes the actual
+	// fields behind "default", if selected. If it has never been set,
+	// the pointer will be nil.
+	AliasMetadata []string `json:"alias_metadata"`
+
+	// fields is a list of the configured default and available
+	// fields. It's intentionally not jsonified.
+	fields *Fields
+}
+
+// GetAliasMetadata is intended to be used on config reads.
+// It gets an explicit list of all the user's configured
+// fields that are being added to alias metadata. It will never
+// include the "default" parameter, and instead includes the actual
+// fields behind "default", if selected.
+func (h *handler) GetAliasMetadata() []string {
+	if h.AliasMetadata == nil {
+		return h.fields.Default
+	}
+	return h.AliasMetadata
+}
+
+// ParseAliasMetadata is intended to be used on config create/update.
+// It takes a user's selected fields (or lack thereof),
+// converts it to a list of explicit fields, and adds it to the handler
+// for later storage.
+func (h *handler) ParseAliasMetadata(data *framework.FieldData) error {
+	userProvided, ok := data.GetOk(h.fields.FieldName)
+	if !ok {
+		// Nothing further to do here.
+		return nil
+	}
+
+	// uniqueFields protects against weird edge cases like if
+	// a user provided "default,field1,field2,default".
+	uniqueFields := make(map[string]bool)
+	for _, field := range userProvided.([]string) {
+		if field == "default" {
+			// Add the fields that "default" represents, rather
+			// than the explicit field.
+			for _, dfltField := range h.fields.Default {
+				uniqueFields[dfltField] = true
+			}
+		} else {
+			// Make sure they've sent a supported field so we can
+			// error early if not.
+			if !strutil.StrListContains(h.fields.all(), field) {
+				return fmt.Errorf("%q is not an available field, please select from: %s", field, strings.Join(h.fields.all(), ", "))
+			}
+			uniqueFields[field] = true
+		}
+	}
```

Comment:
This is doing more than uniquing the field list: uniquing fields, expanding `default` to the default field list, and ensuring the fields are only from the available set of fields. What about splitting these out into individual operations? It looks like `strutil` has several functions we can use:
```go
fields := strutil.RemoveDuplicates(userProvided.([]string))
fields = expandDefaultField(fields) // TODO
ok := strutil.StrListSubset(h.fields.all(), fields)
if !ok {
  // unsupported fields found
}
```

Reply:
That came out nicely. I should lean into that `strutil` package more, it has some nice functions. https://github.com/hashicorp/vault/pull/8783/commits/4022bd1e164e98576266a6d23747e50a4250b321